classes.py

```diff
@@ -0,0 +1,544 @@
+                                                                   row[11])})  # POINTS AGAINST OPPONENTS
+                if len(row[1]) > 0:
+                    self.cbx_to_fexerj[int(row[1])] = int(row[0])
+
+    def load_manual_entry_dict(self):
+        if os.path.exists('manual_entry_list.json'):
+            with open('manual_entry_list.json', 'r') as manual_entry_dict_file:
+                self.manual_entries = json.load(manual_entry_dict_file)
+
+    def write_manual_entry_dict(self):
+        with open('manual_entry_list.json', 'w') as manual_entry_dict_file:
+            manual_entry_dict_file.write(json.dumps(self.manual_entries))
+
+
+class FexerjPlayer:
+
+    def __init__(self, id_fexerj, id_cbx, title, name, lastrating, club, birthday, sex, federation, totgames,
+                 sumopprating, ptsagainstopp):
+        self.id_fexerj = id_fexerj
+        self.id_cbx = id_cbx
+        self.title = title
+        self.name = name
+        self.lastRating = lastrating
+        self.club = club
+        self.birthday = birthday
+        self.sex = sex
+        self.federation = federation
+        self.totGames = totgames
+        self.sumOppRating = sumopprating
+        self.ptsAgainstOpp = ptsagainstopp
+
+
+class TournamentPlayer:
+
+    def __init__(self, tournament, url):
+
+        self.snr = 0
+        self.name = ""
+        self.id = 0
+        self.opponents = []
+        self.tournament = tournament
+        self.load_player_page(url)
+
+        self.is_unrated = None
+        self.is_temp = None
+
+        self.last_k = None
+        self.last_rating = None
+        self.last_total_games = None
+        self.last_sum_oppon_ratings = None
+        self.last_pts_against_oppon = None
+
+        # self.this_rating = None
+        self.this_pts_against_oppon = None
+        self.this_sum_oppon_ratings = None
+        self.this_avg_oppon_rating = None
+        self.this_games = None
+        self.this_score = None
+        self.this_expected_points = None
+        self.this_points_above_expected = None
+
+        self.new_rating = None
+        self.new_total_games = None
+        self.new_sum_oppon_ratings = None
+        self.new_pts_against_oppon = None
+
+    def load_player_page(self, url):
+
+        soup = BeautifulSoup(requests.get(url).content, 'lxml')
+        tables = soup.find_all("table", class_="CRs1")
+
+        if len(tables) == 0:
+            del self
+            return
+
+        table = tables[0]
+        rows = table.find_all("tr")
+        for i in range(0, len(rows)):
+            cells = rows[i].find_all("td")
+            cell_data = [cell.get_text().strip() for cell in cells]
+            if cell_data[0] == "Name":
+                self.name = cell_data[1]
+            elif cell_data[0] == "Starting rank":
+                self.snr = cell_data[1]
+            elif cell_data[0] == "Ident-Number":
+                self.id = int(cell_data[1])
+                if not self.id:
```

Comment:
Is id 0 the default? I'm worried whether the cast above would fail and break everything